### DISCUSSION TITLE
How to add label (title) for y axis?

### DISCUSSION THREAD

**Discussion by xichquylk** [patrykandpatrick/vico#687] [Questions] [answered]
I has a line chart as below, how can I add label for y axis?

![image](https://github.com/patrykandpatrick/vico/assets/32903810/3908e531-fde0-413b-a51a-30a29a697179)

More question:  How can I start x value from begin of x axis?
![image](https://github.com/patrykandpatrick/vico/assets/32903810/34268d1b-1575-4f82-ba3f-6940ad9333f5)

Thanks.

**Comment by Gowsky** [comment, marked as answer]
Hello! You can add an axis title like shown in Chart3 [in the sample app](https://github.com/patrykandpatrick/vico/blob/7941b6f056041dc667a3b38d9a47a1e6e2b6fbed/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart3.kt#L100-L107). You can also find there [an example](https://github.com/patrykandpatrick/vico/blob/7941b6f056041dc667a3b38d9a47a1e6e2b6fbed/sample/src/main/res/values/chart_3_styles.xml#L78) of adding the axis title using XML attributes if you're using the view system. However, the axis title is not positioned like in the picture you provided. If you need to use this exact position, you have to subclass `VerticalAxis` and do the required overrides.

For your second question, use `HorizontalLayout.fullWidth()`, like shown in `Chart3` [in the sample app](https://github.com/patrykandpatrick/vico/blob/7941b6f056041dc667a3b38d9a47a1e6e2b6fbed/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart3.kt#L128). For the view system [see here](https://github.com/patrykandpatrick/vico/blob/7941b6f056041dc667a3b38d9a47a1e6e2b6fbed/sample/src/main/res/layout/chart_3.xml#L30).

**Reply by xichquylk** [reply]
Thanks so much!

**Reply by xichquylk** [reply]
Could you please help me how to override subclass VerticalAxis?

**Reply by Gowsky** [reply]
Just like any other class in Kotlin. [See here](https://kotlinlang.org/docs/inheritance.html).
